submissions/aymkin/OOP-frogger/app.js

```diff
@@ -0,0 +1,121 @@
+let scoreCount = 0;
+
+const START = {
+	x: 202,
+	y: 405,
+};
+
+const FIELD = {
+	column: 101,
+	row: 83,
+	leftBorder: 101,
+	rightBorder: 303,
+	topBorder: 0,
+	bottomBorder: 322,
+};
+
+const SPEED = (min = 75, max = 200) =>
+	Math.floor(Math.random() * (max - min + 1)) + min;
+
+const Creature = function (x, y) {
+	this.x = x;
+	this.y = y;
+};
+
+
+	const Enemy = function(x, y, speed, gamer) {
+	Creature.call(this, x, y);
+	this.speed = speed;
+	this.gamer = gamer;
+	this.sprite = 'images/enemy-bug.png';
+};
+
+Enemy.prototype.update = function(dt) {
+	this.x = this.x + this.speed * dt;
+	if (this.x >= FIELD.rightBorder + FIELD.column * 2) {
+		this.x = -101;
+		this.speed = SPEED();
+	}
+	if (this.isCollision()) {
+		alert(`You lost! The score is demotioned to ${scoreCount -= 1}`);
+		this.gamer.beginAgain();
+	}
+};
+
+Enemy.prototype.render = function() {
+	ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
```

Comment:
The `render ()` method for our classes is the same, we can declare it in the general class.

Reply:
done